frontend/src/core/user/components/UserAutoUpdater.js

```diff
@@ -0,0 +1,60 @@
+/* @flow */
+/* globals IntervalID */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { actions } from '..';
+
+
+type Props = {
+};
+
+type InternalProps = {
+    ...Props,
+    dispatch: Function,
+};
+
+type State = {|
+    timer: ?IntervalID,
+|};
+
+
+/**
+ * Regularly fetch user data to keep it up-to-date with the server.
```

Comment:
Since we don't have this functionality in the existing translate view, I'm curiouis what's the use case for this?

Reply:
That's a very good question. The reason is that on processeer, I have problems with the session expiring quite quickly, but the local state doesn't know about it. So, it regularly happens that I want to save a change and can't because I'm logged out. I thus wanted to make sure that our data about the user stays up-to-date (to a certain extent), so that we don't give them a false experience (like, a Save button when they're actually logged out of the server).

However, Pontoon sessions run a lot longer, and users are very rarely disconnected. It is possible that this feature is useless. What do you think?